x-pack/plugins/fleet/server/services/actions/actions.ts

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { partition } from 'lodash';
+import { v4 as uuidV4 } from 'uuid';
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { fromKueryExpression, toElasticsearchQuery } from '@kbn/es-query';
+import type { estypes } from '@elastic/elasticsearch';
+
+import { FleetActionsError } from '../../../common/errors';
+import { AGENT_ACTIONS_INDEX, AGENT_ACTIONS_RESULTS_INDEX } from '../../../common';
+import { auditLoggingService } from '../audit_logging';
+
+import { appContextService } from '../app_context';
+
+import type { FleetAction, FleetActionResponse } from './types';
+
+export const createAction = async (
+  esClient: ElasticsearchClient,
+  action: Partial<FleetAction>
+): Promise<Partial<FleetAction>> => {
```

Comment:
This should return the full `FleetAction` no a `Partial`

Reply:
7bd6c021b31f6571570bdcef9fcc504be761b1b3